exchanges/kraken/kraken_websocket.go

```diff
@@ -1383,43 +1333,60 @@ func (k *Kraken) wsAddOrder(request *WsAddOrderRequest) (string, error) {
 	return resp.TransactionID, nil
 }
 
-// wsCancelOrders cancels one or more open orders passed in orderIDs param
+// wsCancelOrders cancels open orders concurrently
+// It does not use the multiple txId facility of the cancelOrder API because the errors are not specific
 func (k *Kraken) wsCancelOrders(orderIDs []string) error {
+	wg := sync.WaitGroup{}
+	wg.Add(len(orderIDs))
+	errC := make(chan error, len(orderIDs))
+	for _, id := range orderIDs {
+		go func(id string) {
+			defer wg.Done()
+			if err := k.wsCancelOrder(id); err != nil {
+				errC <- err
+			}
```

Comment:
```suggestion
			errC <- k.wsCancelOrder(id)
```

Reply:
My first reaction:
> I think I'd prefer to avoid sending nils down the channel just to either ignore them at 1356 or have wasteful calls to  AppendError

Then I realised when I said "Wasteful calls to AppendError" I realised I must be being defensive, because I've recently argued for that.

And then I looked at it and realised this is a pattern I've done too many times. So:

Fixed 2ed96411

I'll write tests now for it, but wanted to sling it up for any fast rejections

Reply:
Already approved my man, whatever you want to do. 🤷 😄